Firestore/core/test/firebase/firestore/local/leveldb_index_manager_test.mm

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/test/firebase/firestore/local/index_manager_test.h"
+
+#import "Firestore/Example/Tests/Local/FSTPersistenceTestHelpers.h"
+#import "Firestore/Source/Local/FSTPersistence.h"
+
+#include "Firestore/core/src/firebase/firestore/local/leveldb_index_manager.h"
+#include "absl/memory/memory.h"
+#include "gtest/gtest.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+namespace firebase {
+namespace firestore {
+namespace local {
+
+namespace {
+
+inline id<FSTPersistence> PersistenceFactory() {
```

Comment:
Nit: `inline` is unnecessary here. (It rarely matters for performance. It's more commonly used to avoid violating ODR when having a function definition in a header file, which doesn't apply here).

Reply:
I think you can make a stronger statement: `inline` doesn't matter for performance at all. The function of the `inline` keyword in modern C++ is *only* for avoiding problems with multiple definitions via headers.

Reply:
Okay, I removed it, but FWIW I copied this from elsewhere and we have lots of usage of inline it seems... 😄 
```
~/src/firebase-ios-sdk/Firestore/core (mikelehen/collection-group-queries) $ grep -ir inline . | wc -l
     155
```

Reply:
Note that most of those are in header files, where `inline` _is_ useful to avoid violating ODR (one-definition-rule; basically to avoid linker complaining about duplicate function names).